File: binarySearch.c
// Program to implement Linear Search in C Language
#include <stdio.h>
#include <conio.h>

// Function to implement Binary search
int binarySearch(int arr[], int size, int element)
{
    int low,mid,high;
    // Lowest Index
    low = 0;
    // Highest Index
    high = size -1;

    // When low and high converges loop ends
    while(low<=high)
    {
        // find the mid point and look for the element
        mid = (low+high)/2;
        // if mid == element return the mid index
        if(arr[mid]==element)
        {
            return mid;
        }
        // if element at mid is less than the element
        if(arr[mid]<element)
        {
            // low becomes mid point + 1
            low = mid + 1;
        }
        else
        {
            // else high becomes mid point - 1
            high = mid -1;
        }
    }
    // If element is not found return -1
    return -1;
}

int main()
{
    // Sorted Array for binary search
    int arr[] = {1,3,5,56,64,73,123,225,444};
    // Calculate size of the array
    int size = sizeof(arr)/sizeof(int);
    int element = 64;
    // Call the binary Search function to search element from the sorted array
    int searchIndex = binarySearch(arr, size, element);
    // Print the element and index
    printf("The element %d is found at index %d", element, searchIndex);
    return 0;
}

File: Binary serach implementation in C, C Sharp (C#), JS and Ruby/binary_search.cs
// Program to implement Binary Search in C# (C sharp)
// Please note: We can use visual studio 2022 to run this program by creating a new console app project and copying the code below.

using System;

namespace BinarySearchProgram
{
	class Program
	{
		// Function to implement Binary Search
		static int BinarySearch(int[] arr, int element)
		{
			// Initialize low and high 
			int low = 0;
			int high = arr.Length - 1;

			// Loop until low is less than or equal to high
			while (low <= high)
			{
				// Calculate the mid index
				int mid = (low + high) / 2;

				// Check if the element is at mid then returning mid index
				if (arr[mid] == element)
				{
					return mid;
				}
				// If element at mid is less than the target element, search the right half
				if (arr[mid] < element)
				{
					low = mid + 1;
				}
				// Else search the left half
				else
				{
					high = mid - 1;
				}
			}
			// If the element is not found, return -1
			return -1;
		}
		static void Main(string[] args) {
			
			// Sorted array for Binary Search
			int[] arr = { 1, 3, 5, 56, 64, 73, 123, 225, 444 };
			int element = 73;
			// Call the Binary Search function
			int searchedIndex = BinarySearch(arr, element);

			// Print the result
			if (searchedIndex != -1) {
				Console.WriteLine($"The element {element} found at index {searchedIndex}");
			}
			else
			{
				Console.WriteLine("Element not found in the array");
			}
            
		}
	}
}